src/l1_fetcher.rs

```diff
@@ -34,6 +36,50 @@ pub enum L1FetchError {
     GetTx,
 }
 
+#[derive(Default)]
+struct L1Metrics {
+    // Metrics variables.
+    l1_blocks_processed: u64,
+    l2_blocks_processed: u64,
+    latest_l1_block_nbr: u64,
+    latest_l2_block_nbr: u64,
+
+    first_l1_block: Option<u64>,
+    last_l1_block: Option<u64>,
```

Comment:
Does this need to be an `Option` anymore?

Reply:
Probably not. I'll clean it out :slightly_smiling_face: